tracing-journald/src/lib.rs

```diff
@@ -268,6 +298,60 @@ where
     }
 }
 
+/// The facility argument is used to specify what type of program is logging the
+/// message. This lets the configuration file specify that messages from
+/// different facilities will be handled differently.
+#[derive(Copy, Clone)]
+pub enum SyslogFacility {
+    /// security/authorization messages
+    Auth,
+    /// security/authorization messages (private)
+    Authpriv,
+    /// clock daemon (cron and at)
+    Cron,
+    /// system daemons without separate facility value
+    Daemon,
+    /// ftp daemon
+    Ftp,
+    /// kernel messages (these can't be generated from user
+    /// processes)
+    Kern,
+    ///reserved for local use
+    Local0,
+    ///reserved for local use
+    Local1,
+    ///reserved for local use
+    Local2,
+    ///reserved for local use
+    Local3,
+    ///reserved for local use
+    Local4,
+    ///reserved for local use
+    Local5,
+    ///reserved for local use
+    Local6,
+    ///reserved for local use
+    Local7,
+    /// line printer subsystem
+    Lpr,
+    /// mail subsystem
+    Mail,
+    /// USENET news subsystem
+    News,
+    /// messages generated internally by syslogd(8)
+    Syslog,
+    /// generic user-level messages
+    User,
+    /// UUCP subsystem
+    Uucp,
```

Comment:
i believe that an enum variant's value can be assigned to a constant of the enum's `repr` type (i had to check, see [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c30df5916693ff153929341b4b68a11e)). i think we should be able to write something like this:

```suggestion
#[repr(i32)]
pub enum SyslogFacility {
    /// security/authorization messages
    Auth = libc::LOG_AUTH as i32,
    /// security/authorization messages (private)
    Authpriv = libc::LOG_AUTHPRIV as i32,,
    /// clock daemon (cron and at)
    Cron = libc::LOG_CRON as i32,
    /// system daemons without separate facility value
    Daemon = libc::LOG_DAEMON as i32,
    /// ftp daemon
    Ftp = libc::LOG_FTP as i32,
    /// kernel messages (these can't be generated from user
    /// processes)
    Kern = libc::LOG_KERN as i32,
    ///reserved for local use
    Local0 = libc::LOG_LOCAL0 as i32,
    ///reserved for local use
    Local1 = libc::LOG_LOCAL1 as i32,
    ///reserved for local use
    Local2 = libc::LOG_LOCAL2 as i32,
    ///reserved for local use
    Local3  = libc::LOG_LOCAL3 as i32,
    ///reserved for local use
    Local4 = libc::LOG_LOCAL4 as i32,
    ///reserved for local use
    Local5  = libc::LOG_LOCAL5 as i32,
    ///reserved for local use
    Local6  = libc::LOG_LOCAL6 as i32,
    ///reserved for local use
    Local7 = libc::LOG_LOCAL7 as i32,
    /// line printer subsystem
    Lpr  = libc::LOG_LPR as i32,
    /// mail subsystem
    Mail  = libc::LOG_MAIL as i32,
    /// USENET news subsystem
    News  = libc::LOG_NEWS as i32,
    /// messages generated internally by syslogd(8)
    Syslog  = libc::LOG_SYSLOG as i32,
    /// generic user-level messages
    User = libc::LOG_USER as i32,
    /// UUCP subsystem
    Uucp = libc::LOG_UUCP as i32,
```

if we did this, we could remove the `match` in `put_facility` and replace it with just `let value = facility as i32;`, which would be nice. however, this does rely on these values being a `c_int` (castable to an `i32`) on all platforms' libcs, so we should make sure it wouldn't cause any problems to do this.